sql/core/src/main/scala/org/apache/spark/sql/execution/subquery.scala

```diff
@@ -118,6 +224,19 @@ case class PlanSubqueries(sparkSession: SparkSession) extends Rule[SparkPlan] {
         ScalarSubquery(
           SubqueryExec(s"subquery${subquery.exprId.id}", executedPlan),
           subquery.exprId)
+      case subquery: expressions.Exists =>
```

Comment:
I'm a little confused. After adding the new physical subqueries, I think we just need to update `RewritePredicateSubquery`.

Reply:
Hmmm... Do you mean that we don't need the rule `RewriteUncorrelatedSubquery`? I think it can reduce the result set of the subquery by pushing the computations to `Executor` side instead of `Driver` side.

Reply:
We need `RewriteUncorrelatedSubquery`, but it should be a separated PR.

Reply:
Ok, I will open anther one for it after this PR, and remove it here.